### ISSUE TITLE
"Release snap for pull request" action broken

### ISSUE THREAD

**Issue by sbe-arg** [nextcloud-snap/nextcloud-snap#2966]
We need a better process for testing github actions so we don't break releases.

Ie. Don't merge actions upgrades along nextcloud upgrades

**Comment by pachulo** [comment]
I don't think anything broke with the latest upgrade of an action, but we will see in the following PRs we open.

**Comment by pachulo** [comment]
It seems like everything went fine over here: https://github.com/nextcloud-snap/nextcloud-snap/pull/2973

**Comment by sbe-arg** [comment]
What is very confusing is the failures each time it runs the launchdpad action.

And because we can't publicly see the runner logs I cannot help trobleshoot.

Actions should be always be green if they are doing what is supposed to do. But unsure why we have so much release reds.

When we released 30.0.2 was red for the 30.0.2 tag run and there was no new version available in snap until a few days after.

Whats the release process did you re run something manually after? Or worker but ended in red fail anyway.

**Comment by pachulo** [comment]
> What is very confusing is the are failures each time it runs the launchdpad action.

The [deploy job](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/.github/workflows/launchpad.yml#L14) from the [Deploy to launchpad action](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/.github/workflows/launchpad.yml) has a known problem, because we can't automatically push git tags to Launchpad repos (am I right @kyrofa ?).
So every time there's a new tag to be pushed, it fails:
<img width="1102" alt="image" src="https://github.com/user-attachments/assets/7bdd6c5c-ea7f-488e-bb40-e567451a398c">

So we have to push them manually.

> Actions should be always be green if they are doing what is supposed to do. But unsure why we have so much release reds.

I would say is a mix of things, but is true that a lot of [Release snap for pull request actions](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/.github/workflows/release-pr-snap.yml) runs are failing lately:
<img width="1102" alt="image" src="https://github.com/user-attachments/assets/9ce8c047-e40f-4d1b-bd26-1b1f65b58edd">

It's because the [Upload snap step](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/.github/workflows/release-pr-snap.yml#L111) never finishes:
<img width="1102" alt="image" src="https://github.com/user-attachments/assets/b46458e4-8d05-4dec-9511-3460c0123ecf">

Do you have any idea why @kyrofa ?

> When we released 30.0.2 was red for the 30.0.2 tag run and there was no new version available in snap until a few days after.

In [that specific case](https://github.com/nextcloud-snap/nextcloud-snap/commit/d596fc22d9b415c7f05c3db591f0240631e4a0a1) the red was on the Snap Publisher action, but for that action we don't have Details, because...it's implemented by scripts on the [Release snap for pull request workflow](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/.github/workflows/release-pr-snap.yml)?
Anyway, I don't really know why the action was executed on that merge commit 😓 

> Whats the release process did you re run something manually after? Or worked but ended in red fail anyway.

No, there's manual work involved. [Here](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Release-Management) you have all the details of the release process.

**Comment by sbe-arg** [comment]
Ill have a deeper read at the link and see if I can help further

**Comment by pachulo** [comment]
I forgot that @kyrofa already wrote about the logic for creating channels to upload the PR built snaps: https://kyrofa.com/posts/new-and-improved-way-to-review-github-pull-requests-for-snaps/

**Comment by kyrofa** [comment]
> because we can't automatically push git tags to Launchpad repos (am I right @kyrofa ?).

Yes you _were_ right, but I think github supports what we need now. I tweaked the settings, let's see if that works.

> I've just seen that most of the times they fail [with this error](https://github.com/nextcloud-snap/nextcloud-snap/blob/master/.github/workflows/release-pr-snap.yml#L61):
> `Unhandled error: Error: Expected one pull request to contain commit, but got 0`

I haven't created a PR for a bit, so I haven't received any failure notifications. First time I've seen this error. I'm not sure what's happening there... maybe something has changed in the API. But then it looks like it succeeds sometimes too, so maybe we have a logic error. We'll need to look into it.

**Comment by kyrofa** [comment]
> And because we can't publicly see the runner logs I cannot help trobleshoot.

This is news to me. You can't see our action logs? I thought being signed in was the only requirement.

**Comment by sbe-arg** [comment]
> > And because we can't publicly see the runner logs I cannot help trobleshoot.
> 
> This is news to me. You can't see our action logs? I thought being signed in was the only requirement.

We need to be a member to see the logs.

**Comment by pachulo** [comment]
I don't know if this is related, but it seems like the snap store is struggling these days:
1. The latest snaps on the `*/edge` channels are from 4 days ago. They are being built OK, but fail to upload with this error: <img width="247" alt="image" src="https://github.com/user-attachments/assets/0c36ff92-6802-48a1-9a49-2631a06ecd80">

2. The Launchapd-built snaps for the latests commits on our `28`, `29`, and `develop` git branches also failed to upload: <img width="127" alt="image" src="https://github.com/user-attachments/assets/16d90ddf-feb9-451b-958c-6b0b4085ac1e">


Anything we should worry about @kyrofa ?

**Comment by pachulo** [comment]
> Huh, what does our listing page look like?

The `Releases` tab looks like this:
<img width="1400" alt="image" src="https://github.com/user-attachments/assets/926c95a3-df6f-463b-a883-d7f022d518e1">

By the way, the `Releases` tab is very sluggish on my side; I even get a `504 Gateway Time-out`
 error sometimes. But everything seems to be fine on [their status page](https://status.snapcraft.io/)... 🤔 Maybe we just have way too many releases?

Can you check if you experience the same @kyrofa ?

**Comment by kyrofa** [comment]
> I even get a 504 Gateway Time-out error sometimes

Yes, it's always been that way. We have more releases than the web UI knows how to handle. Even when I knew the people maintaining that software I couldn't get them to fix it. I don't have a lot of confidence it'll get fixed.

Anyway, yeah, it looks like we're caught in a review queue somehow, but the web UI doesn't give me any obvious insight into why. I think I got all the way back to the original, and check this out:

![Screenshot from 2024-11-25 13-44-37](https://github.com/user-attachments/assets/343ea4e6-4bc4-475c-ab99-5dc57bc5492e)

Maybe their automatic review tool had a hiccup, and eventually passed, but none of the other snaps enqueued for review were ever picked back up again. I took the liberty of just rejecting it on our side... let's see if the review queue catches back up.

**Comment by pachulo** [comment]
> Confirmed, looks like the review queue is slooooowly catching up now (each review takes a few minutes, and we have a few hundred to go).

Everything looks nicer now indeed.

> Yes you were right, but I think github supports what we need now. I tweaked the settings, let's see if that works.

It seems that what you did work and now tags are also pushed to Launchpad! 
<img width="465" alt="image" src="https://github.com/user-attachments/assets/9ae28dc7-45c1-4523-b201-2204358a5e98">
What was it?

**Comment by pachulo** [comment]
> It seems that what you did work and now tags are also pushed to Launchpad!

Well, it's not working as expected: the tags pushed from the github action are not annotated; look at [this example](https://git.launchpad.net/nextcloud-snap/tag/?h=30.0.2snap2) and compare it to the latest [one I pushed manually](https://git.launchpad.net/nextcloud-snap/tag/?h=30.0.2snap1).

How can we solve that?

**Comment by kyrofa** [comment]
Looks to me like we should probably be explicitly pushing tags in that "deploy" workflow.

**Comment by kyrofa** [comment]
Ah, no I bet it's more related to this: https://github.com/actions/checkout/issues/882

**Comment by pachulo** [comment]
The fetch-tags thing from https://github.com/actions/checkout/issues/882#issuecomment-2254187250 didn't work: the 30.0.4snap1 tag was pushed without annotation 😢 

I will be trying https://github.com/actions/checkout/issues/882#issuecomment-1231911465 once I have more time

**Comment by pachulo** [comment]
> I will be trying https://github.com/actions/checkout/issues/882#issuecomment-1231911465 once I have more time

OK, this workaround worked! 😄 

By the way, what's the status of this issue @sbe-arg ? You can't still see the logs?

**Comment by sbe-arg** [comment]
I can now... we can probably close it.

**Comment by scubamuc** [comment]
> I can now... we can probably close it.
